### PR TITLE
tile-converter: i3s - convert COLOR_0

### PR DESCRIPTION
**Pull request by belom88**
London: 

Screenshot from 2021-02-11 12-14-27
[3DTILES_London_ubuntu_modules_cli_slpk_draco_full_bundle_16_v17.log.zip](https://github.com/visgl/loaders.gl/files/5965208/3DTILES_London_ubuntu_modules_cli_slpk_draco_full_bundle_16_v17.log.zip)

Berlin:

Screenshot from 2021-02-11 12-03-11
[3DTILES_Berlin_ubuntu_modules_cli_slpk_draco_full_bundle_16_v17.log.zip](https://github.com/visgl/loaders.gl/files/5965209/3DTILES_Berlin_ubuntu_modules_cli_slpk_draco_full_bundle_16_v17.log.zip)

Munich:

Screenshot from 2021-02-11 12-23-28
[3DTILES_Munich_ubuntu_modules_cli_slpk_draco_full_bundle_16_v17.log.zip](https://github.com/visgl/loaders.gl/files/5965211/3DTILES_Munich_ubuntu_modules_cli_slpk_draco_full_bundle_16_v17.log.zip)

### CODE REVIEW

File: modules/tile-converter/src/i3s-converter/helpers/geometry-converter.js [hunk 1]
```diff
@@ -116,7 +116,7 @@ async function _makeNodeResources({
     ? await generateCompressedGeometry(vertexCount, convertedAttributes, {
         positions,
         normals,
-        texCoords,
+        texCoords: texture ? texCoords : new Float32Array(0),
```
Comment (ibgreen) [inline modules/tile-converter/src/i3s-converter/helpers/geometry-converter.js:119]: Nit: Is it better to initialize to an empty array or to `null`

File: modules/tile-converter/src/i3s-converter/helpers/geometry-converter.js [hunk 2]
```diff
@@ -400,6 +396,32 @@ function flattenTexCoords(texCoords, indices) {
   }
   return newTexCoords;
 }
+
+/**
+ * Convert color from COLOR_0 based on indices to plain arrays, compatible with i3s
```
Comment (ibgreen) [inline modules/tile-converter/src/i3s-converter/helpers/geometry-converter.js:401]: Can this comment be written more clearly.

Comment (ibgreen) [reply modules/tile-converter/src/i3s-converter/helpers/geometry-converter.js:401]: You could write a test for this function.

**Review by ibgreen — APPROVED** [review]

**Comment by ibgreen** [comment]
Does this display correctly also in the loaders.gl examples? If not we should fix that in separate PR.

**Comment by Tamrat-B** [comment]
@belom88 this is looking good! It looks like though normals from the 3d tile content are not coming thru (though, few of the i3s nodes I checked do have normals (i.e., its not flat); also the color saturation (look at the lake) is different between 3D tiles and i3s, see below (the converted 3d tiles in i3s, left, the 3d tiles in cesium, on the right):
image

**Pull request merged**